Framework/PythonInterface/plugins/algorithms/LoadElementalAnalysisData.py

```diff
@@ -0,0 +1,179 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+
+import glob
+import os
+import numpy as np
+from mantid import config
+from mantid.api import AlgorithmFactory, AnalysisDataService, PythonAlgorithm, TextAxis, WorkspaceGroup, WorkspaceGroupProperty
+from mantid.kernel import Direction, IntBoundedValidator
+from mantid.simpleapi import ConvertToHistogram, ConvertToPointData, DeleteWorkspace, LoadAscii, WorkspaceFactory
+
+
+type_keys = {"10": "Prompt", "20": "Delayed", "99": "Total"}
+spectrum_index = {"Delayed": 1, "Prompt": 2, "Total": 3}
+num_files_per_detector = 3
+run_number_length = 5
+
+
+class LoadElementalAnalysisData(PythonAlgorithm):
+    def category(self):
+        """
+        Returns category
+        """
+        return "DataHandling"
+
+    def PyInit(self):
+        self.declareProperty(name='Run', defaultValue=0, validator=IntBoundedValidator(lower=0),
```

Comment:
Why not to use FileProperty or MultipleFileProperty instead? Those would do the searching, validation, full path resolution and file algebra for you.

Reply:
We construct the files from the runs, because there is a significant risk that the user would miss something.

Reply:
I wonder if this points at a hole in FileProperty/MultipleFileProperty that might help the ILL cases too Gagik?

The property currently has no way of specifying any kind of template for how the filename should look such that it can then be filled in with information from the user. That might be helpful to support more complex file finding without having to reinvent the logic?

Reply:
Note that this is of course outside the scope of this PR!

Reply:
@AnthonyLim23 Can you bring an example of constructed file name from run?
@martyngigg It might be useful, however the multiple file parsing and validation is already very complex and delicate, it would be a great effort to not break it. Btw, I opened this issue, if you have an idea please comment:  https://github.com/mantidproject/mantid/issues/29881

Reply:
they are of the form ral<run_number>.rooth<ID>.dat, where the <ID> is a numerical code that signifies the detector and spectrum (total, delayed or prompt).

Reply:
Sorry, what is the code? the run_number?

Reply:
We give it the run number and it will construct the file names, using the convention above. We do not know beforehand which detectors or ID's are present.

Reply:
Ok, then it's justified.